### PR TITLE
Do not create two loggers for DeprecationLogger backport(#58435) 

### PR DESCRIPTION
**Pull request by pgomulka**
DeprecationLogger's constructor should not create two loggers. It was
taking parent logger instance, changing its name with a .deprecation
prefix and creating a new logger.
Most of the time parent logger was not needed. It was causing Log4j to
unnecessarily cache the unused parent logger instance.

depends on https://github.com/elastic/elasticsearch/pull/61515
backports #58435

### CODE REVIEW

File: modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/ScriptProcessor.java [hunk 1]
```diff
@@ -53,7 +52,7 @@
 public final class ScriptProcessor extends AbstractProcessor {
 
     private static final DeprecationLogger deprecationLogger =
-            new DeprecationLogger(LogManager.getLogger(DynamicMap.class));
+            DeprecationLogger.getLogger(DynamicMap.class);
```
Comment (pgomulka) [inline modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/ScriptProcessor.java:55]: there is a series of classes that possibly incorrectly use DynamicMap logger
This PR grew so large and was mostly automatic (find/replace, backport, automated refactoring) that I am reluctant to fix it here too. Maybe I miss something and `DynamicMap` was actually intended here?
@pugnascotia WDYT?

Comment (pugnascotia) [reply modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/ScriptProcessor.java:55]: Looks like a mistake to me. @stu-elastic what do you think?

File: modules/percolator/src/main/java/org/elasticsearch/percolator/PercolateQueryBuilder.java [hunk 2]
```diff
@@ -100,7 +99,7 @@
 public class PercolateQueryBuilder extends AbstractQueryBuilder<PercolateQueryBuilder> {
     public static final String NAME = "percolate";
 
-    private static final DeprecationLogger deprecationLogger = new DeprecationLogger(LogManager.getLogger(ParseField.class));
+    private static final DeprecationLogger deprecationLogger = DeprecationLogger.getLogger(ParseField.class);
```
Comment (pugnascotia) [inline modules/percolator/src/main/java/org/elasticsearch/percolator/PercolateQueryBuilder.java:102]: Looks like another mistake.

File: server/src/main/java/org/elasticsearch/ingest/ConditionalProcessor.java [hunk 3]
```diff
@@ -47,8 +46,7 @@
 
 public class ConditionalProcessor extends AbstractProcessor implements WrappingProcessor {
 
-    private static final DeprecationLogger deprecationLogger =
-            new DeprecationLogger(LogManager.getLogger(DynamicMap.class));
+    private static final DeprecationLogger deprecationLogger = DeprecationLogger.getLogger(DynamicMap.class);
```
Comment (pugnascotia) [inline server/src/main/java/org/elasticsearch/ingest/ConditionalProcessor.java:49]: This one looks just plain wrong.

File: server/src/main/java/org/elasticsearch/rest/action/document/RestMultiTermVectorsAction.java [hunk 4]
```diff
@@ -39,8 +38,7 @@
 import static org.elasticsearch.rest.RestRequest.Method.POST;
 
 public class RestMultiTermVectorsAction extends BaseRestHandler {
-    private static final DeprecationLogger deprecationLogger = new DeprecationLogger(
-        LogManager.getLogger(RestTermVectorsAction.class));
+    private static final DeprecationLogger deprecationLogger = DeprecationLogger.getLogger(RestTermVectorsAction.class);
```
Comment (pugnascotia) [inline server/src/main/java/org/elasticsearch/rest/action/document/RestMultiTermVectorsAction.java:41]: This one is incorrect too.

File: server/src/main/java/org/elasticsearch/script/AbstractSortScript.java [hunk 5]
```diff
@@ -36,8 +35,7 @@
 
 abstract class AbstractSortScript implements ScorerAware {
 
-    private static final DeprecationLogger deprecationLogger =
-            new DeprecationLogger(LogManager.getLogger(DynamicMap.class));
+    private static final DeprecationLogger deprecationLogger = DeprecationLogger.getLogger(DynamicMap.class);
```
Comment (pugnascotia) [inline server/src/main/java/org/elasticsearch/script/AbstractSortScript.java:38]: This one too.

File: server/src/main/java/org/elasticsearch/script/AggregationScript.java [hunk 6]
```diff
@@ -40,8 +39,7 @@
 
     public static final ScriptContext<Factory> CONTEXT = new ScriptContext<>("aggs", Factory.class);
 
-    private static final DeprecationLogger deprecationLogger =
-            new DeprecationLogger(LogManager.getLogger(DynamicMap.class));
+    private static final DeprecationLogger deprecationLogger = DeprecationLogger.getLogger(DynamicMap.class);
```
Comment (pugnascotia) [inline server/src/main/java/org/elasticsearch/script/AggregationScript.java:42]: And this one.

File: server/src/main/java/org/elasticsearch/script/FieldScript.java [hunk 7]
```diff
@@ -39,8 +38,7 @@
 
     public static final String[] PARAMETERS = {};
 
-    private static final DeprecationLogger deprecationLogger =
-            new DeprecationLogger(LogManager.getLogger(DynamicMap.class));
+    private static final DeprecationLogger deprecationLogger = DeprecationLogger.getLogger(DynamicMap.class);
```
Comment (pugnascotia) [inline server/src/main/java/org/elasticsearch/script/FieldScript.java:41]: And this one.

File: server/src/main/java/org/elasticsearch/script/ScoreScript.java [hunk 8]
```diff
@@ -65,8 +64,7 @@ public Explanation get(double score, Explanation subQueryExplanation) {
         }
     }
 
-    private static final DeprecationLogger deprecationLogger =
-            new DeprecationLogger(LogManager.getLogger(DynamicMap.class));
+    private static final DeprecationLogger deprecationLogger = DeprecationLogger.getLogger(DynamicMap.class);
```
Comment (pugnascotia) [inline server/src/main/java/org/elasticsearch/script/ScoreScript.java:67]: And this one.

File: server/src/main/java/org/elasticsearch/script/ScriptedMetricAggContexts.java [hunk 9]
```diff
@@ -66,8 +65,7 @@ public Object getState() {
 
     public abstract static class MapScript {
 
-        private static final DeprecationLogger deprecationLogger =
-                new DeprecationLogger(LogManager.getLogger(DynamicMap.class));
+        private static final DeprecationLogger deprecationLogger = DeprecationLogger.getLogger(DynamicMap.class);
```
Comment (pugnascotia) [inline server/src/main/java/org/elasticsearch/script/ScriptedMetricAggContexts.java:68]: And this one.

**Review by pugnascotia — APPROVED** [review]

**Pull request merged**